src/sentry/api/endpoints/organization_stats_v2.py

```diff
@@ -0,0 +1,239 @@
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=organization, user=request.user)
+        for team in team_list:
+            project_list.extend(Project.objects.get_for_user(team=team, user=request.user))
+        project_ids = list({p.id for p in project_list})
+
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["project_id", "category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [organization.id], "project_id": project_ids},
+            orderby=["times_seen", "time"],
+        )
+
+        response = {project_id: StatsResponse(start, end, rollup) for project_id in project_ids}
+        for row in result:
+            stat_to_update = response[row["project_id"]].get(row["category"])
+            stat_to_update.update(row)
+
+        return Response(
+            {project_id: stat_res.build_fields() for project_id, stat_res in response.items()}
+        )
+
+
+class OrganizationProjectStatsDetails(ProjectEndpoint, OrganizationEndpoint):
+    def get(self, request, project):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+
+        # TODO: see if there's a better way to get the user's projects
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=project.organization, user=request.user)
+        for team in team_list:
+            project_list.extend(Project.objects.get_for_user(team=team, user=request.user))
+        project_ids = list({p.id for p in project_list})
+
+        if project.id not in project_ids:
+            # TODO: more info in response here?
+            return Response(status=404)
+
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [project.organization.id], "project_id": [project.id]},
+            orderby=["time"],
+        )
+        response = StatsResponse(start, end, rollup)
+        for row in result:
+            stat_to_update = response.get(row["category"])
+            stat_to_update.update(row)
+
+        return Response(response.build_fields())
+
+
+# TODO: good names for these classes
+class StatsResponse:
+    def __init__(self, start, end, rollup):
+        self.errors = TimeSeriesValues(start, end, rollup)
+        self.transactions = TimeSeriesValues(start, end, rollup)
+        self.attachments = TimeSeriesValues(start, end, rollup)
+
+    _GETTERS = {
+        DataCategory.ERROR: (lambda s: s.errors),
+        DataCategory.TRANSACTION: (lambda s: s.transactions),
+        DataCategory.ATTACHMENT: (lambda s: s.attachments),
+    }
```

Comment:
I take it this is based on the pattern I used in the `UsageStat` class in `billing_usage.py`? It's really only worth it if you want to expose the regular attributes while also offering the `get` method (in my case, I started with the attributes and was forced to bolt on the `get` afterward).

If you can get by with only the `get` (which seems to be the case here at a glance), I'd recommend skipping the attributes entirely and just having an internal dict, keyed by the categories. That simplifies a few things:

```python
    def __init__(self, start, end, rollup):
        self._values = {k: TimeSeriesValues(start, end, rollup) for k in CATEGORY_NAME_MAP}

    def get(self, category):
        return self._values[category]

    def __iter__(self):
        return iter(self._values.items())
```
etc.